### PR TITLE
Drop dependency on `py` and simplify tests

### PR DESCRIPTION
**Pull request by frenzymadness**
The first commit drops dependency on py module
and switch from LocalPath to pathlib.Path which
has a different API so it required some more changes.

The second commit actually removes the unnecessary subdirs.
If you want to keep them, I can drop the second commit.

### CODE REVIEW

File: tests/test_common.py [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 from databooks.common import find_obj
 
 
-def test_find_obj(tmpdir: LocalPath) -> None:
+def test_find_obj(tmp_path: Path) -> None:
     """Find file based on name, and search path."""
     filename = "SAMPLE_FILE.ext"
 
-    start_dir = Path(tmpdir)
+    start_dir = Path(tmp_path)
```
Comment (tverum) [inline tests/test_common.py:10]: Unneccesary cast

File: tests/test_common.py [hunk 2]
```diff
@@ -19,11 +17,11 @@ def test_find_obj(tmpdir: LocalPath) -> None:
     assert filepath.is_file()
 
 
-def test_find_obj__missing(tmpdir: LocalPath) -> None:
+def test_find_obj__missing(tmp_path: Path) -> None:
     """Return `None` when looking for file along path."""
     filename = "SAMPLE_FILE.ext"
 
-    start_dir = Path(tmpdir)
+    start_dir = Path(tmp_path)
```
Comment (tverum) [inline tests/test_common.py:24]: Redundant cast

**Comment by murilo-cunha** [comment]
Thanks for the PR! 🚀  Could you rebase on `main`?
I've changed the CICD a bit so we ran the tests (linting, ...) on PRs as well

**Comment by frenzymadness** [comment]
> Thanks for the PR! rocket Could you rebase on `main`? I've changed the CICD a bit so we ran the tests (linting, ...) on PRs as well

Done, but you have to approve the workflow.

**Comment by frenzymadness** [comment]
Linter problem fixed.

**Review by murilo-cunha — APPROVED** [review]
Thanks!!! 🚀 🎉

**Pull request merged**